### ISSUE TITLE
@:strict is completely ignored when applied to a var field.

### ISSUE THREAD

**Issue by Obi-Wan-Clemobi** [HaxeFoundation/haxe#5665]
@:strict is completely ignored when applied to a var field.

I am trying to use @:strict to add an annotation (JCIP's GuardedBy annotation) to a var field for Java. Ultimately, the annotation could be anything, but that is my specific use case. 
I am able to do this for function fields and class type but @:strict seems to be ignored and stripped from the var field.

I created an extern for net.jcip.annotations.GuardedBy and applied it where applicable.

Example:

``` haxe
package com.company.annotations;
#if java
@:native("net.jcip.annotations.GuardedBy") extern class GuardedBy
{
    public function value() : String;
}
#end
```

Usage:

``` haxe
@:strict(com.company.annotations.GuardedBy({ value :"lock" }))
private var toBeGuarded:String;          // Does not retain the annotation!
@:strict(com.company.annotations.GuardedBy({ value :"lock" }))
public function doSomething() : Void
{
     // Some code where lock is used
}
```

**Comment by wiggin77** [comment]
I can confirm @meta is ignored for vars as well.  I need to add native meta tags to vars for Java and C#.

**Comment by Simn** [comment]
The Java target will be put in "regression fixes only" state with the Haxe 4.3.0 release. See https://github.com/HaxeFoundation/haxe/wiki/Target-Support-Level#java-and-c-since-430 for additional information. We apologize for any inconvenience this causes!
